pkg/bootstrap/docker/dockerhelper/helper.go

```diff
@@ -23,43 +25,27 @@ const openShiftInsecureCIDR = "172.30.0.0/16"
 
 // Helper provides utility functions to help with Docker
 type Helper struct {
-	client *docker.Client
+	client          *docker.Client
+	engineAPIClient *dockerclient.Client
 }
 
 // NewHelper creates a new Helper
-func NewHelper(client *docker.Client) *Helper {
+func NewHelper(client *docker.Client, engineAPIClient *dockerclient.Client) *Helper {
 	return &Helper{
-		client: client,
+		client:          client,
+		engineAPIClient: engineAPIClient,
 	}
 }
 
 type RegistryConfig struct {
 	InsecureRegistryCIDRs []string
 }
 
-func getRegistryConfig(env *docker.Env) (*RegistryConfig, error) {
-	for _, entry := range *env {
-		if !strings.HasPrefix(entry, "RegistryConfig=") {
-			continue
-		}
-		glog.V(5).Infof("Found RegistryConfig entry: %s", entry)
-		value := strings.TrimPrefix(entry, "RegistryConfig=")
-		config := &RegistryConfig{}
-		err := json.Unmarshal([]byte(value), config)
-		if err != nil {
-			glog.V(2).Infof("Error unmarshalling RegistryConfig: %v", err)
-			return nil, err
-		}
-		glog.V(5).Infof("Unmarshalled registry config to %#v", config)
-		return config, nil
-	}
-	return nil, nil
-}
-
-func hasCIDR(cidr string, listOfCIDRs []string) bool {
+func hasCIDR(cidr string, listOfCIDRs []*registry.NetIPNet) bool {
 	glog.V(5).Infof("Looking for %q in %#v", cidr, listOfCIDRs)
 	for _, candidate := range listOfCIDRs {
-		if candidate == cidr {
+		candidateStr := (*net.IPNet)(candidate).String()
```

Comment:
not that I can tell, it's going to be an array returned by docker info, and registry.NetIPNet is a net.IPNet.